main.go

```diff
@@ -1,29 +1,54 @@
 package main
 
 import (
-	"context"
-	"flag"
-	"log"
+        "context"
+        "flag"
+        "log"
 
-	"github.com/equinix/terraform-provider-equinix/equinix"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
+        "github.com/equinix/terraform-provider-equinix/equinix"
+        "github.com/equinix/terraform-provider-equinix/internal/provider"
+        "github.com/equinix/terraform-provider-equinix/version"
+
+        "github.com/hashicorp/terraform-plugin-framework/providerserver"
+        "github.com/hashicorp/terraform-plugin-go/tfprotov5"
+        "github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server"
+        "github.com/hashicorp/terraform-plugin-mux/tf5muxserver"
 )
 
 //go:generate go run github.com/hashicorp/terraform-plugin-docs/cmd/tfplugindocs
 func main() {
+
+	ctx := context.Background()
+
 	var debugMode bool
 
 	flag.BoolVar(&debugMode, "debug", false, "set to true to run the provider with support for debuggers like delve")
 	flag.Parse()
-	opts := &plugin.ServeOpts{ProviderFunc: equinix.Provider}
+
+	providers := []func() tfprotov5.ProviderServer{
+			providerserver.NewProtocol5(
+					provider.CreateFrameworkProvider(version.ProviderVersion)),
+			equinix.Provider().GRPCProvider,
+	}
+
+	muxServer, err := tf5muxserver.NewMuxServer(ctx, providers...)
+	if err != nil {
+			log.Fatal(err)
+	}
+
+	var serveOpts []tf5server.ServeOpt
 
 	if debugMode {
-		err := plugin.Debug(context.Background(), "registry.terraform.io/equinix/equinix", opts)
-		if err != nil {
-			log.Fatal(err.Error())
-		}
-		return
+			serveOpts = append(serveOpts, tf5server.WithManagedDebug())
 	}
 
-	plugin.Serve(opts)
-}
+	err = tf5server.Serve(
+			"registry.terraform.io/equnix/equinix",
```

Comment:
```suggestion
			"registry.terraform.io/equinix/equinix",
```

Reply:
If tests didn't / can't uncover this being wrong, we may want to see some manually testing to confirm that this provider works with a combination of framework and sdkv2 plugins.

Reply:
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create
 <= read (data resources)

Terraform will perform the following actions:

  # data.equinix_metal_project_ssh_key.foobar will be read during apply
  # (config refers to values not yet known)
 <= data "equinix_metal_project_ssh_key" "foobar" {
      + created     = (known after apply)
      + fingerprint = (known after apply)
      + id          = (known after apply)
      + name        = (known after apply)
      + owner_id    = (known after apply)
      + project_id  = "b3b111d4-a313-4132-a824-XXXX"
      + public_key  = (known after apply)
      + search      = (known after apply)
      + updated     = (known after apply)
    }

  # equinix_metal_project_ssh_key.foobar will be created
  + resource "equinix_metal_project_ssh_key" "foobar" {
      + created     = (known after apply)
      + fingerprint = (known after apply)
      + id          = (known after apply)
      + name        = "ocm-tf-test"
      + owner_id    = (known after apply)
      + project_id  = "b3b111d4-a313-4132-a824-f48a06cf7ca3"
      + public_key  = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDM/unxJeFq..."
      + updated     = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + test = (known after apply)

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

equinix_metal_project_ssh_key.foobar: Creating...
equinix_metal_project_ssh_key.foobar: Creation complete after 0s [id=651002ea-fbeb-4969-a3e6-XXXX]
data.equinix_metal_project_ssh_key.foobar: Reading...
data.equinix_metal_project_ssh_key.foobar: Read complete after 0s [id=651002ea-fbeb-4969-a3e6-XXXX]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

test = "48:b8:72:58:d5:32:36:45:f3:4d:b5:17:ef:82:50:e3"
```

here's some references in the output log to the tmux provider 
![image](https://github.com/equinix/terraform-provider-equinix/assets/68897552/0d94aa5e-05d4-430b-abe5-d75075615387)